packages/form-js-editor/src/features/palette/components/Palette.js

```diff
@@ -184,4 +195,46 @@ function getIndefiniteArticle(type) {
   }
 
   return 'a';
+}
+
+/**
+ * Returns a list of palette entries.
+ *
+ * @param {FormFields} formFields
+ * @returns {Array<PaletteEntry>}
+ */
+export function collectPaletteEntries(formFields) {
+  return Object.entries(formFields._formFields).map(([ type, formField ]) => {
+
+    const { config: fieldConfig } = formField;
+
+    return {
+      label: fieldConfig.label,
+      type: type,
+      group: fieldConfig.group,
+      icon: fieldConfig.icon,
+      iconUrl: fieldConfig.iconUrl
+    };
+  }).filter(({ type }) => type !== 'default');
+}
+
+/**
+ * There are various options to specify an icon for a palette entry.
+ *
+ * a) via `iconUrl` property in a form field config
```

Comment:
I was wondering if there could be any security issues to consider about this, [and apparently there isn't](https://stackoverflow.com/a/17090136)
But if we ever consider allowing custom components in the web modeler we have to consider [CSP policies](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) (we would probably have to host the icons ourselves)

Reply:
![image](https://github.com/bpmn-io/form-js/assets/9433996/0d8a304a-02bf-4d35-93f7-e1d6cb6b014d)